src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKeyProviderBuilder.java

```diff
@@ -0,0 +1,128 @@
+package com.amazonaws.encryptionsdk.kms;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.RegionUtils;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.kms.AWSKMSClient;
+import com.amazonaws.services.kms.AWSKMSClientBuilder;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Used to build a {@link KmsMasterKeyProvider} using {@link AWSKMSClientBuilder}, which will use provider chains to get
+ * defaults if the properties are not explicitly set.
+ *
+ * Default usage will use all the defaults found by AWSKMSClientBuilder. If a region is not found in the configuration
+ * chain, then {@link Regions#DEFAULT_REGION} is used.
+ * <pre>
+ *     KmsMasterKeyProvider keyProvider = KmsMasterKeyProviderBuilder.defaultProvider();
+ * </pre>
+ *
+ * Example, specifying a region:
+ * <pre>
+ *     KmsMasterKeyProvider keyProvider = KmsMasterKeyProviderBuilder.standard()
+ *       .withRegion("us-east-1")
+ *       .build();
+ * </pre>
+ */
+public class KmsMasterKeyProviderBuilder {
```

Comment:
This is already missing withMetricsCollector() and withRequestHandlers(). Does it make sense to simply inherit from AwsClientBuilder so that you always get the appropriate (and new) configuration as available?  That said, I'm worried that you might need to use reflection (possibly Proxies, which would be ugly) to properly plumb all current and future calls to your inner client builder.

At the very least, please add withMetricsCollector() and withRequestHandlers() and consider more future proof solutions.